runner/ollamarunner/runner.go

```diff
@@ -883,6 +890,11 @@ func (s *Server) completion(w http.ResponseWriter, r *http.Request) {
 					return
 				}
 
+				if !grammarSet && grammar != nil && tokenParser.ConstraintsAllowed() {
+					seq.sampler.SetGrammar(grammar)
```

Comment:
You probably could just call SetGrammar every time and avoid `grammarSet` and `grammar != nil`. It possibly depends on the next comment but there's no real cost to SetGrammar right now.